src/cc/ws/SimFederate.java

```
@@ -0,0 +1,121 @@
+	{
+		ServletConfig oConf = getServletConfig();
+
+		String sRetry = oConf.getInitParameter("retry");
+		if (sRetry != null)
+			m_lRetryInterval = Integer.parseInt(sRetry) * 1000L;
+
+		String sAddress = oConf.getInitParameter("ambassador");
+		if (sAddress != null)
+		{
+			try
+			{
+				InetAddress.getByName(sAddress); // validate network address
+				m_sAmbassadorAddress = sAddress;
+				m_oTs = new TimeSource(true);
+
+				String sUrl = oConf.getInitParameter("url");
+				if (sUrl != null)
+					m_sCarmaCloudUrl = sUrl;
+
+				new Thread(this).start(); // begin registration cycle
+			}
+			catch (Exception oEx)
+			{
+			}
+		}
+
+		if (m_oTs == null) // simulation time source not created
+			m_oTs = new TimeSource(false);
+	}
+
+
+	@Override
+	public void doPost(HttpServletRequest oReq, HttpServletResponse oRes)
+	   throws ServletException, IOException
+	{
+		JSONObject oJson;
+		try (BufferedInputStream oIn = new BufferedInputStream(oReq.getInputStream()))
+		{
+			oJson = new JSONObject(new JSONTokener(oIn));
+		}
+
+		long lSeq = -1L;
+		long lStep = -1L;
+		if (oJson.has("seq") && oJson.has("timestep"))
+		{
+			lSeq = oJson.getLong("seq");
+			lStep = oJson.getLong("timestep");
+			m_oTs.m_lSimTime = lStep;
+		}
+		LOGGER.debug(String.format("seq: %d timestep: %d timesource: %d", lSeq, lStep, m_oTs.currentTimeMillis()));
+	}
+
+
+	@Override
+	public void run()
+	{
+		boolean bRegistered = false;
+		while (!bRegistered)
+		{
+			try (Socket oSock = new Socket())
+			{
+				oSock.connect(new InetSocketAddress(m_sAmbassadorAddress, 1617), 10000);
+				DataOutputStream oOut = new DataOutputStream(oSock.getOutputStream());
+				oOut.writeUTF(String.format("{\"id\":\"carma-cloud\", \"url\":\"%s\"}", m_sCarmaCloudUrl));
+				bRegistered = true;
+			}
+			catch (Exception oEx)
+			{
+				LOGGER.error(oEx);
+			}
+
+			if (!bRegistered)
+			{
+				try
+				{
+					Thread.sleep(m_lRetryInterval);
+				}
+				catch (Exception oEx)
+				{
+				}
```

This is an empty catch statement that will essentially silence and exceptions thrown by thread sleep

---

Now handles both illegal argument and interrupt exceptions.

---

This still looks like an empty catch statement. Am I missing something here?